src/Resources/js/modules/participation-pre-toggle-handler.ts

```diff
@@ -27,36 +27,57 @@ export class ParticipationPreToggleHandler {
                 return;
             }
 
-            if (self.needUserInteractionBeforeToggle($checkbox)) {
-                let participantCounter: ParticipantCounter = $checkbox.data(ParticipantCounter.NAME);
-                if (participantCounter.isAvailable()) {
-                    self.executeBeforeToggle($checkbox);
-                } else {
-                    CombinedMealOffersService.execute($checkbox, self.participationToggleHandler);
-                }
-            } else if (self.isUnbookedCombinedDish($checkbox)) {
-                let $mealContainer = $checkbox.closest('.meal');
-                let simpleDishSlugs = self.getSimpleDishSlugs($mealContainer);
-                if (0 === simpleDishSlugs.length) {
-                    console.log('combined-meal dishes not found');
-                    return;
-                }
-                let data: SerializedFormData[] = [];
-                simpleDishSlugs.forEach(function (slug, i) {
-                    data.push({
-                        'name': `dishes[${i}]`,
-                        'value': slug
-                    });
-                });
-                self.participationToggleHandler.toggle($checkbox, data);
+            if (self.isUnbookedCombinedDish($checkbox)) {
+                self.processCombinedMealJoin($checkbox);
             } else {
                 self.participationToggleHandler.toggle($checkbox);
             }
         });
     }
 
-    private needUserInteractionBeforeToggle($checkbox: JQuery): boolean {
-        return this.isUnbookedCombinedDish($checkbox) && this.isCombinedDishWithVariations($checkbox);
+    private processCombinedMealJoin($checkbox: JQuery): void {
+        if (this.existDishVariations($checkbox)) {
```

Comment:
we could remove the else like in php

if () {
  this.joi...
  return;
}

this.joi...